src/ILVerify/src/ILImporter.Verify.cs

```diff
@@ -1276,7 +1266,21 @@ void ImportStoreIndirect(int token)
 
         void ImportStoreIndirect(TypeDesc type)
         {
-            throw new NotImplementedException($"{nameof(ImportStoreIndirect)} not implemented");
+            ClearPendingPrefix(Prefix.Unaligned);
+            ClearPendingPrefix(Prefix.Volatile);
+
+            var value = Pop();
+            var address = Pop();
+
+            Check(!address.IsReadOnly, VerifierError.ReadOnlyIllegalWrite);
+
+            CheckIsByRef(address);
+            if (type != null)
+                CheckIsAssignablePointer(type, address.Type);
+            else
+                CheckIsObjRef(address.Type);
+
+            CheckIsAssignable(value, address.DereferenceByRef());
```

Comment:
I think that this should check `value` -> `type` assignability. Consider this:

```
  .method private hidebysig static void  Main() cil managed
  {
    .entrypoint
    .maxstack  2
    .locals init (object V_0)

    ldloca.s   V_0
    ldstr      "Hello"
    stobj  [mscorlib]System.IO.Stream
    ret
  }
```

PEVerify will fail on this, but this check will not.